File: tslib-py/tslib/analyze.py
from typing import List

from ._http import _Session
from .models import AnalyzeResult


class AnalyzeAPI:
    def __init__(self, session: _Session) -> None:
        self._s = session

    def analyze(
        self,
        data: List[float],
        *,
        k: int = 1,
        n: int = 10,
        window_size: int = 5,
    ) -> AnalyzeResult:
        body = {"data": data, "k": k, "n": n, "windowSize": window_size}
        return AnalyzeResult._from(self._s.post("/api/analyze", body))

File: tslibR/R/analyze.R
#' Analyze a time series
#'
#' Computes descriptive statistics, ACF/PACF, an ADF stationarity test, and
#' common transforms (log, first difference, rolling average) in one call.
#'
#' @param client A [`tslib_client`] object.
#' @param data Numeric vector of observations.
#' @param k Integer lag for autocovariance (default 1).
#' @param n Number of ACF/PACF lags to return (default 10).
#' @param window_size Rolling-average window size (default 5).
#' @return Named list with statistics, ACF/PACF arrays, ADF result, and
#'   pre-computed transforms.
#' @export
tslib_analyze <- function(client, data, k = 1L, n = 10L, window_size = 5L) {
  body <- list(data = as.list(data), k = k, n = n, windowSize = window_size)
  .tslib_post(client, "/api/analyze", body)
}
